File: dashboard_backend/app/schemas/ec2.py
from typing import Literal

from pydantic import BaseModel


class Ec2FindingResponse(BaseModel):
    instance: str
    region: str
    instanceType: str
    issue: str
    cpuAverage: str
    cpuP95: str
    memoryAverage: str
    recommendation: str
    estimatedSaving: float
    risk: Literal["Low", "Medium", "High", "Critical"]
    status: str

File: dashboard/src/types/ec2.ts
export type RiskLevel = "Low" | "Medium" | "High" | "Critical"

export type Ec2Finding = {
  instance: string
  region: string
  instanceType: string
  issue: string
  cpuAverage: string
  cpuP95: string
  memoryAverage: string
  recommendation: string
  estimatedSaving: number
  risk: RiskLevel
  status: string
}
